lib/p2p/Pool.ts

```diff
@@ -358,6 +358,8 @@ class Pool extends EventEmitter {
 
     for (const address of sortedAddresses) {
       if (node.lastAddress && addressUtils.areEqual(address, node.lastAddress)) continue;
+      // check if its tor node address
+      if (!this.config.tor && address.host.indexOf('.onion') !== -1) continue;
```

Comment:
This logic looks good, but the check is only happening for when we connect to peer's advertised addresses. It doesn't happen when we connect to the last good address, or when we use the Connect rpc command.

I think this logic should go in the `addOutbound` method which is called on every connection. In there it should throw an error if we don't allow tor addresses but the address is for tor. You'll have to create a new error in the p2p errors.ts file to be thrown. That error will then also need to be added to the switch statement in grpc/getGrpcError.ts - I figure it makes most sense to be added as an `ILLEGAL_ARGUMENT` error. That's so that if someone attempts to connect to a tor address over rpc when doing so is disabled, they'll get an intelligible error message.

Let me know if you have any questions.

Reply:
Yeah, you really think a lot about all the functions, errors and comments texts - it takes even more time than implementing something)

Reply:
status.ILLEGAL_ARGUMENT error doesn't exist, maybe you meant INVALID_ARGUMENT?